### PR TITLE
Spinward Smoothies Space Ruin

### PR DESCRIPTION
**Pull request by Sylphet**
## About The Pull Request

Adds a space ruin called Spinward Smoothies. It's a nature-themed smoothie bar built into an asteroid. Also includes recipes for six smoothies to match the theme.

image

image

## Why It's Good For The Game

It's a cute little thing to find in space, enabling space explorers to grab a drink there and relax for a while. More things to explore.

## Changelog

:cl:
add: Added Spinward Smoothies Space Ruin
add: Added six types of smoothies
/:cl:

### CODE REVIEW

**Comment by Sylphet** [comment]
heated CI moment

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -1203,3 +1203,70 @@
 	glass_name = "glass of töchtaüse syrup"
 	glass_desc = "Not for drinking on its own."
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+
```
Comment (optimumtact) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:1207]:
```suggestion
/datum/reagent/consumable/citrus_control
	name = "citrus control smoothie"
	description = "A powerfully citrus flavored smoothie that can cause delusions of grandeur."
	color = "#FFA500"
	nutriment_factor = 0
	taste_description = "orange and lime"
	glass_icon_state = "orange_and_lime"
	glass_name = "citrus control smoothie"
	glass_desc = "A classic drink, can be very controversial though as it divides the world on those who enjoy the taste and those who do not."
	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
```

Comment (EOBGames) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1207]: No headcoder references in my land

**Review by san7890 — CHANGES_REQUESTED** [review]
Hey there,

Thanks for the strong work in this map, just a few things that should be addressed.

First things first... where are the icons? You display a new billboard with the icon state `"billboard_smoothies"` in `'icons/obj/billboard.dmi'`... but that DMI has not been changed in this PR. This causes this to show up in my editor:

image

Above all else, please rectify this to ensure everything works properly. More images/reviews are in the spoiler section below.

<details>
<summary>Click For Images</summary>

image

On both of these vendors, the `onstation` variable should be set to "0" or "FALSE", since they are not on station. This is just a code assumption.

image

Terracotta Siding abruptly stops here- would probably look better to have it persist as a wrap around the entire bar area. Let me know if that's a design choice.

image
These rocks here... don't really make too much sense. They sort-of impede progress, and since the rest of the smoothie store is rather nice- it doesn't particularly make sense for the main entrance way to this establishment to have inhibitory rock outcroppings. I also think that it shouldn't be a sort of "boring" walkway into the store... work that problem, and let me know if you need help.

I also think there should also be a more explicit "entrance", although I think having the marker beacons mark the path is a nice touch.

image

This airlock feels rather bland to me- however you can avoid one altogether. Since it's a ruin that you don't exactly want to completely vent out, you can use a `/obj/structure/fans/tiny` to block _all_ atmospherics changes under an airlock. You can trim it down to just one airlock in that instance, if that helps you design a bit better.

FYI: If you want to commit to a full "airlock" experience, you're missing cyclelink helpers (`/obj/effect/mapping_helpers/airlock/cyclelink_helper`). Look at a station map to find out how we implement it if you're not sure.

image
Please don't mount lights right on asteroid rock, if the rock is broken (which is easy to do), it looks ugly.

image

Does a smoothie shop _really_ need this? Especially with the fact that this place has unlimited power... not really great. Maybe a fridge with all of the smoothie making stuff (or a cold room)?

image

Since this is a ruin, you don't want this to be explicitly space (for a number of reasons). Can you please convert all the space tiles on here to have both `/turf/template_noop` and `/area/template_noop`? Look at other space ruins to see that implementation.

image

All your asteroid sand turfs are the standard variant, when they should be `/turf/open/misc/asteroid/airless`. Otherwise, we get Active Turfs (not good).

image
This should also be an airless variant.
</details>

I haven't compiled the map to do a more in-depth review, just wanted to get the major issues out of the way (design and other technical stuff). I've marked this PR as a draft for internal management, feel free to mark it as "Ready for Review" when this has been addressed. Thank you for your good work.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (cacogen) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:1207]: it should be lemon and lime, in reference to the best maintainer

**Review by san7890 — CHANGES_REQUESTED** [review]
For some reason CI is failing with the following error:

```
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:32:error: undefined type: /obj/structure/flora/ausbushes/brflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:35:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:40:error: undefined type: /obj/structure/flora/ausbushes/ppflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:42:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:44:error: undefined type: /obj/structure/flora/rock/pile/largejungle
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:46:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:57:error: undefined type: /obj/structure/flora/ausbushes/ppflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:60:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:90:error: undefined type: /obj/structure/flora/ausbushes/brflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:93:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:139:error: undefined type: /obj/structure/flora/ausbushes/brflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:141:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:158:error: undefined type: /obj/structure/flora/ausbushes/ppflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:160:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:162:error: undefined type: /obj/structure/flora/ausbushes/ywflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:164:error: unknown type
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:195:error: undefined type: /obj/structure/flora/ausbushes/ywflowers
_maps/RandomRuins/SpaceRuins/spinwardsmoothies.dmm:198:error: unknown type
```

Please look into that, and mark this PR "Ready for Review" when addressed.

EDIT: It is most likely related to the merging of #66978.

**Comment by Sylphet** [comment]
Can't believe I'm getting *codegriefed*

**Review by san7890 — CHANGES_REQUESTED** [review]
Alright, final pass:

<details>
<summary>Click For More Images</summary>

image

These are all on space, you want these to all be on Turf Passthrough.

image

This can also happen (I removed the catwalks to better illustrate this) that since they're a part of the ruin, the lighting blends to the rest of the ruin (AKA: horribly). You'll want to use the `/area/space/nearstation` definition on all your catwalks (including the one of the billboard) instead of the "area passthrough" definition to have it blend with space lighting.

image

In fact, it might also look better if you make the whole asteroid rock have the nearstation area, since it has a weird clashing effect to observers. It's alright if you don't really want to do this... but I think it would make the whole thing look a bit better.

image

Could you use the `/obj/effect/turf_decal/siding/wideplating/terracotta/end` (and then have the direction shifted as you want it) here instead of having two separate decals?

---

This is something that I'm addressing in case you weren't aware of it, feel free to discard it if you're OK with what it looks like in game (the following image):
image

However, in the map, you have that same area plotted out as the following in StrongDMM:
image

You deliberately chose certain flooring to show up, but that gets smoothed out in game. If you like the effect, you can keep it, but you could also set the variable `smoothing_flags` on the tiles (turfs) you don't want to smooth to `0` in order to preserve the mapping vision.
</details>

Anyways, that should wrap it up from that point. Thank you!

**Pull request merged**